### PR TITLE
Update the default class group hash parameters

### PR DESCRIPTION
**Pull request by jonas-lj**
* Update the default parameters for the class group hash to both keeping it fast + ensure security.
* Update test vectors accordingly.
* Small updates to the code, mainly for readbility.

The  test vectors in sui-framework must be updated after this is merged.

### CODE REVIEW

File: fastcrypto-vdf/benches/class_group.rs [hunk 1]
```diff
@@ -43,18 +41,15 @@ fn class_group_ops(c: &mut Criterion) {
 }
 
 fn qf_from_seed_single<M: Measurement>(discriminant_string: &str, group: &mut BenchmarkGroup<M>) {
-    for k in [1, 2, 4, 8, 16, 32, 64] {
-        let discriminant =
-            Discriminant::try_from(BigInt::from_str_radix(discriminant_string, 10).unwrap())
-                .unwrap();
+    let discriminant =
+        Discriminant::try_from(BigInt::from_str_radix(discriminant_string, 10).unwrap()).unwrap();
```
Comment (kchalkias) [inline fastcrypto-vdf/benches/class_group.rs:45]: ooc, when can this panic?

File: fastcrypto-vdf/benches/class_group.rs [hunk 2]
```diff
@@ -43,18 +41,15 @@ fn class_group_ops(c: &mut Criterion) {
 }
 
 fn qf_from_seed_single<M: Measurement>(discriminant_string: &str, group: &mut BenchmarkGroup<M>) {
```
Comment (kchalkias) [inline fastcrypto-vdf/benches/class_group.rs:43]: why do we call this "from_seed" while the seed is thread locally randomly generated? (and probably not even with a secure rng?)

File: fastcrypto-vdf/src/class_group/hash.rs [hunk 3]
```diff
@@ -23,105 +22,90 @@ use crate::math::jacobi;
...
-        let (a, mut b) = sample_modulus(discriminant, seed, k)?;
-
-        // b must be odd but may be negative
-        if b.is_even() {
-            b -= &a;
+    /// This method returns an [InvalidInput] error if the discriminant is so small that there are
+    /// no secure parameters, and it may also happen if the discriminant is not a prime.
+    pub fn hash_to_group(seed: &[u8], discriminant: &Discriminant) -> FastCryptoResult<Self> {
+        // This lower limit ensures that the default, secure parameters give valid results, a reduced quadratic form.
+        if discriminant.bits() < 600 {
```
Comment (kchalkias) [inline fastcrypto-vdf/src/class_group/hash.rs:32]: let's have this "600" as a constant, also explain what does it mean?

File: fastcrypto-vdf/src/class_group/hash.rs [hunk 4]
```diff
@@ -23,105 +22,90 @@ use crate::math::jacobi;
...
-    ///
-    /// This method returns an InvalidInput error if the discriminant is smaller than 800 bits.
-    pub fn hash_to_group_with_default_parameters(
-        seed: &[u8],
-        discriminant: &Discriminant,
-    ) -> FastCryptoResult<Self> {
-        let k = get_default_k(discriminant.bits() as usize);
-        Self::hash_to_group(seed, discriminant, k)
+        // Parameters are chosen such that the image of the hash function is ~260 bits to avoid collision and, and each prime factor is ~130 bits to avoid them being precomputed.
+        hash_to_group_with_custom_parameters(seed, discriminant, 17, 2)
```
Comment (kchalkias) [inline fastcrypto-vdf/src/class_group/hash.rs:36]: same here for 17 and 2, constants and explain why they are set to 17 and 2.

File: fastcrypto-vdf/src/class_group/hash.rs [hunk 5]
```diff
@@ -23,105 +22,90 @@ use crate::math::jacobi;
...
+fn hash_to_group_with_custom_parameters(
+    seed: &[u8],
+    discriminant: &Discriminant,
+    prime_factor_size_in_bytes: u64,
+    prime_factors: u64,
+) -> FastCryptoResult<QuadraticForm> {
+    // Ensure that the image is sufficiently large
+    debug_assert!(prime_factors as f64 * n_bit_primes(prime_factor_size_in_bytes * 8) >= 256f64);
+
+    // Ensure that the prime factors are so large that the corresponding quadratic form be precomputed.
```
Comment (kchalkias) [inline fastcrypto-vdf/src/class_group/hash.rs:58]: "form be procomputed" -> "can" or "cannot" be?

File: fastcrypto-vdf/src/class_group/hash.rs [hunk 6]
```diff
@@ -23,105 +22,90 @@ use crate::math::jacobi;
...
+    let (a, mut b) = sample_modulus(
+        seed,
+        discriminant,
+        prime_factor_size_in_bytes,
+        prime_factors,
+    )?;
+
+    // b must be odd but may be negative
+    if b.is_even() {
+        b -= &a;
```
Comment (kchalkias) [inline fastcrypto-vdf/src/class_group/hash.rs:77]: what does this offer?

File: fastcrypto-vdf/benches/class_group.rs (hunk 1 above)
Comment (jonas-lj) [reply fastcrypto-vdf/benches/class_group.rs:45]: Due to the unwrap, this happens if the given discriminant is not a negative (probable) prime that is 1 mod 8.

File: fastcrypto-vdf/src/class_group/hash.rs (hunk 6 above)
Comment (jonas-lj) [reply fastcrypto-vdf/src/class_group/hash.rs:77]: This picks the right square root to ensure that the quadratic form is valid and reduced.

**Review by kchalkias — APPROVED** [review]

**Pull request merged**